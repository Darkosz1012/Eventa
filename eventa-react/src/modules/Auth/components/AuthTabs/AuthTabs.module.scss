.container{
    width: 500px;
    max-width: 100%;
    // margin: 16px;
    padding: 16px;
    background-color: var(--color-main);
    border-radius: 16px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.527);
}