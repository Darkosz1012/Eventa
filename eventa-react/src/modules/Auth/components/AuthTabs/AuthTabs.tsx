import { Tab, Tabs, useTheme } from "@mui/material"
import React from "react"
import SignInForm from "../SignInForm/SignInForm"
import styles from "./AuthTabs.module.scss"
function AuthTabs(){
    // const theme = useTheme();
    const [activeTab, setActiveTab] = React.useState(0);
    
    return(
        <div className={styles.container}>
            <Tabs value={activeTab} onChange={(_event,value)=>{setActiveTab(value)}}>
                <Tab label="Sign In"/>
                <Tab label="Sign Up"/>
            </Tabs>
            <div hidden={activeTab != 0}>
                <SignInForm></SignInForm>
            </div>
            <div hidden={activeTab != 1}>
                <SignInForm></SignInForm>
            </div>
        </div>
    )
}

export default AuthTabs