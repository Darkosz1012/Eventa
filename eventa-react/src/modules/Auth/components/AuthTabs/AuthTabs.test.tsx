import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import AuthTabs from './AuthTabs.tsx';
import renderer from 'react-test-renderer';

jest.mock('../SignInForm/SignInForm', () => {
    return function DummySignInForm() {
        return (<div>Mocked SignInForm</div>);
    };
});


describe('AuthTabs Component', () => {

    test('mock',()=>{
        const component = renderer.create(<AuthTabs />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('renders AuthTabs component', () => {
        const { getByText } = render(<AuthTabs />);
        expect(getByText('Sign In')).toBeInTheDocument();
        expect(getByText('Sign Up')).toBeInTheDocument();
    });

    test('default active tab should be "Sign In"', () => {
        const { getByText } = render(<AuthTabs />);
        expect(getByText('Sign In')).toHaveClass('Mui-selected');
    });

    test('clicking "Sign Up" tab makes it active', () => {
        const { getByText } = render(<AuthTabs />);
        fireEvent.click(getByText('Sign Up'));
        expect(getByText('Sign Up')).toHaveClass('Mui-selected');
    });
});