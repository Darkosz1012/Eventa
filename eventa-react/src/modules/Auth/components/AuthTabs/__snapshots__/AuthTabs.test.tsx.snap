// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthTabs Component mock 1`] = `
<div
  className="container"
>
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style={
        {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          Sign In
          <span
            className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style={{}}
          />
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          Sign Up
        </button>
      </div>
    </div>
  </div>
  <div
    hidden={false}
  >
    <div>
      Mocked SignInForm
    </div>
  </div>
  <div
    hidden={true}
  >
    <div>
      Mocked SignInForm
    </div>
  </div>
</div>
`;
